core/sql/optimizer/NATable.cpp

```diff
@@ -5485,9 +5485,13 @@ NABoolean createNAFileSets(TrafDesc * table_desc       /*IN*/,
 
    Lng32 postCreateNATableWarnings = CmpCommon::diags()->getNumber(DgSqlCode::WARNING_);
 
+
    if(postCreateNATableWarnings != preCreateNATableWarnings)
      tableConstructionHadWarnings_=TRUE;
-   const char *lobHdfsServer = CmpCommon::getDefaultString(LOB_HDFS_SERVER);
+   char  lobHdfsServer[256] ; // max length determined by dfs.namenode.fs-limits.max-component-length(255)
+   memset(lobHdfsServer,0,256);
+   strncpy(lobHdfsServer,CmpCommon::getDefaultString(LOB_HDFS_SERVER), sizeof(lobHdfsServer));
```

Comment:
OK will add a null terminator or do a strncpy for 255 since we re allready memsetting the array. The length is supposed to be 255 max. We do want to limit it and not allow the CQD setting to be something aritrarily large. If it is we will get a hdfs connection error so it's ok. We do need better error reporting though. WIll address that separately.